src/AppInstallerCLICore/Workflows/MultiQueryFlow.cpp

```diff
@@ -72,15 +72,15 @@ namespace AppInstaller::CLI::Workflow
             switch (m_searchPurpose)
             {
             case SearchPurpose::Install:
-                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(/* isUpgrade */ false);
+                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(m_searchPurpose);
                 break;
             case SearchPurpose::Upgrade:
-                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(/* isUpgrade */ true);
+                searchContext << Workflow::SelectSinglePackageVersionForInstallOrUpgrade(m_searchPurpose);

```

Comment:
nit: The 2 cases SearchPurpose::Install and SearchPurpose::Upgrade can be merged.